src/App.js

```diff
@@ -58,8 +60,23 @@ export const App = () => {
         <Route path="/products" element={<ProductListing />} />
         <Route path="/products/:id" element={<ProductDetail />} />
         <Route path="/search/:query" element={<SearchResultPage />} />
-        <Route path="/wishlist" element={<Wishlist />} />
-        <Route path="/cart" element={<Cart />} />
+        <Route
+          path="/wishlist"
+          element={
+            <PrivateRoute>
+              <Wishlist />
+            </PrivateRoute>
+          }
+        />
+
+        <Route
+          path="/cart"
+          element={
+            <PrivateRoute>
+              <Cart />
+            </PrivateRoute>
+          }
+        />
```

Comment:
Just a suggestion: Instead of wrapping cart and wishlist in `PrivateRoute` component separately. You can do it this way:

`<Route element={ <PrivateRoute/> } >
 <Route path="/wishlist" element={<Wishlist />} />
 <Route path="/cart" element={<Cart />} />
  and so on for private pages...
</Route>
`

Reply:
Ok i will refactor it.